### DISCUSSION TITLE
ReverseEngineering Module

### DISCUSSION THREAD

**Discussion by decoo810** [FreeCAD/FreeCAD#11504] [Discussions that will turn in to issues but need refining]
Hi, does anyone know how to activate "ReverseEngineering" -> "Approximation" -> (all other than 1st and last) ?
Maybe someone who is deep in code can help...
at least what is required as input...
pls help ;-)
Having some help I can offer to prepare some help pages for those functions ;-) 
(BTW, plane creation works nice)

**Comment by sasobadovinac** [comment]
You need a mesh, you can try to generate some basic mesh shapes in the Mesh WB with the "Regular solid.." tool, try for example to create the cylinder mesh with an unchecked "closed" option, then use the Reverse Engineering WB -> Approximation -> Cylinder tool on it...

The general idea of the Reverse Engineering WB is that you would make a 3d scan of something in the real world (laser scanners or photogrammetry techniques) so you would have a mesh (or maybe at first just a point cloud that you would then first translate to a mesh). Then with these tools, you would change that into solids and/or surfaces.
